### PR TITLE
feature: empty alias

### PR DESCRIPTION
**Pull request by Alizter**
This grants #4161.

We add an `@empty` alias allowing the user to build nothing. We also add a check so that user defined rules are not added to the empty alias.

I don't know if we want to or need to be more strict on the checks. For instance, we can bypass the simple rules and add things to the `empty` alias, but it seems pointless to check for this all the time.

- [ ] changelog

### CODE REVIEW

File: test/blackbox-tests/test-cases/alias-empty.t [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  > EOF
+
+This fails as expected
+  $ dune build @empty
+  File "dune", lines 1-4, characters 0-64:
+  1 | (rule
+  2 |  (alias empty)
+  3 |  (action
+  4 |   (echo "I shouldn't be added!")))
+  Error: User-defined rules cannot be added to the 'empty' alias
```
Comment (Alizter) [inline test/blackbox-tests/test-cases/alias-empty.t:25]: TODO: make this error message more in line with other similar errors

**Comment by Alizter** [comment]
OK so there is a weird constraint here. I've restricted user-defined rules from adding things to the `empty` alias however a cram test, like we have in the repo, called `empty.t` will then fail when it generates the `empty` alias.

This is why the CI is failing because we are failing on this restriction after bootstrapping.

@rgrinberg @snowleopard the idea that users should be restricted from adding rules in the `empty` alias seems like a good one, but how can we ensure it doesn't restrict people creating cram tests with those names?

**Comment by rgrinberg** [comment]
Maybe you can change the test specific aliases to just be the full filename of the test starting from the next version of dune? You can still keep the old names as well, but just forbid `empty.t` for tests.

**Comment by rgrinberg** [comment]
Merging this and the restriction on the `@empty` alias can be introduced starting from the next dune version.  We can improve things further over time.

**Comment by maiste** [comment]
This is mostly to not forget about it but shouldn't an entry be added in the reference manual for the new alias?
